### ISSUE TITLE
Does it work now?

### ISSUE THREAD

**Issue by kameronfox** [slooi/NovelAI-Setup-Guide#1]
It  was was working before and it's server just says 503. Any idea dev? Help

**Comment by gigachuds** [comment]
im having the same problems, always 502 bad geteway

**Comment by manaheti1** [comment]
> It was was working before and it's server just says 503. Any idea dev? Help
NovelAI DMCA strike
https://github.com/github/dmca/blob/master/2022/10/2022-10-31-novelai.md

**Comment by kameronfox** [comment]
> > It was was working before and it's server just says 503. Any idea dev? Help
> > NovelAI DMCA strike
> > https://github.com/github/dmca/blob/master/2022/10/2022-10-31-novelai.md

is there anyway to use this any other way?

**Comment by FarroTakeshi** [comment]
> > It was was working before and it's server just says 503. Any idea dev? Help
> > NovelAI DMCA strike
> > https://github.com/github/dmca/blob/master/2022/10/2022-10-31-novelai.md

wow! bad news for us.Thanks for the information

**Comment by manaheti1** [comment]
Nah I was wrong, It maybe was because of pytorch-lightning 1.8.0 (which was released on 1-Nov) (the version we used to use is 1.7.7). I did try to downgrade but it still not work (maybe other modules got updated too, Idk, but it still fixable).

**Comment by kameronfox** [comment]
> Nah I was wrong, It maybe was because of pytorch-lightning 1.8.0 (which was released on 1-Nov) (the version we used to use is 1.7.7). I did try to downgrade but it still not work (maybe other modules got updated too, Idk, but it still fixable).

You are not dev right? it's slooi ? Or are you working on this too? When are you fix it?

**Comment by manaheti1** [comment]
I don't think I can fix this. You should wait for slooi or you can check this (ignore the video check the comment by Shahan Mohammad, they got the similar problem to us, and some people actually got it) https://www.youtube.com/watch?v=TyIL7XjMUaw&lc=UgznDfJ-al5qQlLMKWB4AaABAg 
![image](https://user-images.githubusercontent.com/65333593/199554290-ad40910e-1a1c-4d4f-a59c-2d5fbd14b5d0.png)

**Comment by manaheti1** [comment]
I found the solution:
change pytorch-lightning version to 1.7.7 and scikit-image>=0.19 in requirements.txt file before installing.
**How to do it**:
paste this:
```
import fileinput
import sys

def replaceAll(file,searchExp,replaceExp):
    for line in fileinput.input(file, inplace=1):
        if searchExp in line:
            line = line.replace(searchExp,replaceExp)
        sys.stdout.write(line)
replaceAll("requirements.txt","pytorch_lightning","pytorch_lightning==1.7.7")
replaceAll("requirements.txt","scikit-image","scikit-image>=0.19")
```
between `%cd /content/naifu` and 
`!pip install virtualenv && bash ./setup.sh`  in cell 1
(I can't provide image because my PC prohibit it).
**Extra**: 
In case this repo got shutdown by DMCA: Back up source code of this repo (It likes 4KB)
In case 2 files (naifu.tar and animefull-latest.tar) host expired: Back up them, you can upload it on your own host.

**Comment by kameronfox** [comment]
> I found the solution: change pytorch-lightning version to 1.7.7 and scikit-image>=0.19 in requirements.txt file before installing. **How to do it**: paste this:
> 
> import fileinput import sys
> 
> def replaceAll(file,searchExp,replaceExp): for line in fileinput.input(file, inplace=1): if searchExp in line: line = line.replace(searchExp,replaceExp) sys.stdout.write(line) replaceAll("requirements.txt","pytorch_lightning","pytorch_lightning==1.7.7") replaceAll("requirements.txt","scikit-image","scikit-image>=0.19")
> 
> between `%cd /content/naifu` and `!pip install virtualenv && bash ./setup.sh` in cell 1 (I can't provide image because my PC prohibit it). **Extra**: In case this repo got shutdown by DMCA: Back up source code of this repo (It likes 4KB) In case 2 files (naifu.tar and animefull-latest.tar) host expired: Back up them, you can upload it on your own host.

Ummm... I'm completely new to this coding stuff. Can you be more detailed? Like any video guide would be  helpful.

Step by step tutorial pls.

**Comment by manaheti1** [comment]
It just only paste nothing much:
Step 1: follow main page introduction until Step 3 (Enable GPU) 
![image](https://user-images.githubusercontent.com/65333593/199702800-793a98e1-5e9c-4ccd-aa02-85f04b39cf9f.png)
Step 2: paste the code between %cd /content/naifu and
!pip install virtualenv && bash ./setup.sh
**REMEMBER TO ARRANGE CODE TO LOOK EXACT LIKE THE IMAGE (Python is strict in this problem)**
![image](https://user-images.githubusercontent.com/65333593/199705227-09a949b9-1a10-4d71-8cc1-b0625c52903b.png)
Step 3: follow main page introduction from Step 4 onward
![image](https://user-images.githubusercontent.com/65333593/199706637-2b4eb07b-421d-456a-8284-9e471ac08206.png)

**Comment by kameronfox** [comment]
> It just only paste nothing much: Step 1: follow main page introduction until Step 3 (Enable GPU) ![image](https://user-images.githubusercontent.com/65333593/199702800-793a98e1-5e9c-4ccd-aa02-85f04b39cf9f.png) Step 2: paste the code between %cd /content/naifu and !pip install virtualenv && bash ./setup.sh **REMEMBER TO ARRANGE CODE TO LOOK EXACT LIKE THE IMAGE (Python is strict in this problem)** ![image](https://user-images.githubusercontent.com/65333593/199705227-09a949b9-1a10-4d71-8cc1-b0625c52903b.png) Step 3: follow main page introduction from Step 4 onward ![image](https://user-images.githubusercontent.com/65333593/199706637-2b4eb07b-421d-456a-8284-9e471ac08206.png)

Wow this actually work! Thanks you so much mate!

Thanks for detailed instuction. Well it was super easy XDD. But couldn't figure where to find those texts.

**Comment by hahaxddd18** [comment]
Nice. I've made a backup of the colab and the naifu/animefull-latest archives on my gdrive.  
It seems like it will be gone soon, judging from the other dmca'd repos.

**Comment by kameronfox** [comment]
> Nice. I've made a backup of the colab and the naifu/animefull-latest archives on my gdrive. It seems like it will be gone soon, judging from the other dmca'd repos.

1. Can i have access to that colab? 
2. AAlso if possible can you run it by myself just like you? backing up on my gdrive and run later? If so how to do it?
3. Can you run it on your pc?
4. What is it exactly? T.T i've no idea what it is  other than it generates images, Is it a exe or where it downloads those 7gb file etc.

**Comment by manaheti1** [comment]
**Here is how to backup repo**
![image](https://user-images.githubusercontent.com/65333593/199764153-383b8dc1-2586-47a0-8f7a-faf2957a5bfa.png)
then this
![image](https://user-images.githubusercontent.com/65333593/199764465-1dd23f95-3fd4-4b11-877d-bc5538dac445.png)
Your old repository’s clone URL: URL of this repo: https://github.com/slooi/ImageAI-Setup-Guide/
Repository Name: something that no one will notice
I think you must set repo privacy to public in order to run it on Google collab.

**Link to download 2 files:**
novelAI: https://pub-2fdef7a2969f43289c42ac5ae3412fd4.r2.dev/naifu.tar
model: https://pub-2fdef7a2969f43289c42ac5ae3412fd4.r2.dev/animefull-latest.tar

**Run it on PC**
Yes you can, just execute code of this repo on your PC (on cell 1 you don't need download part since it already on your PC, but you still need Modules for Python (only once)) after that you only needed to run cell 2. But you need a strong GPU.
 
**Second Plan**
The guys who uploaded NovelAI on github that got DMCA striked actually not that sorry.
https://www.youtube.com/watch?v=TyIL7XjMUaw&t=225s
In this video he send us a txt file that got encrypted and a tool to decrypt 
decrypt that txt file into ipynb and we can run his version of novelAICollab.

**If anything unclear, you should ask your IT friends, they can help you with this.**

**Comment by Redir1** [comment]
Hello, is it possible make ai draw better hands? also any way to make nai work with stable 1.5? or it's already updated?

**Comment by manaheti1** [comment]
> Hello, is it possible make ai draw better hands? also any way to make nai work with stable 1.5? or it's already updated?
put "bad hand" into Undesired tag. try looking on reddit NovelAI thread for prompts like "finely detail beautiful hand" like that, maybe it will help.
This version is the leak version and not up to date with current version running on NovelAI official website.
"work with stable 1.5" you mean running diffusionAI with NovelAI models? It just existed yesterday but the result is terrible so the creator delete it.

**Comment by hahaxddd18** [comment]
I didn't make a backup of the github repo as this repo only stores the ipynb file, which you can just make a backup of on google drive itself. It's like 5kb or something. But what the other person did also works fine.
I first used this `https://colab.research.google.com/drive/1LF2a4tY-bnyWJEbUeblv21nVuNDopNzQ` to download the two .tar files onto a brand new google account instead of downloading it and then uploading it manually. That notebook does it for you.  
Save it anywhere, it doesn't matter.
Make the accessibility for naifu.tar and animefull-latest.tar to open so your colab can download it. Don't share it though, the file will get rate limited, just make your own copy.  
Then install  gdown using `!pip install gdown`
Then use gdown to download the file from your drive as following: 
`!gdown https://drive.google.com/u/2/uc?id=xxx `
`!gdown https://drive.google.com/u/2/uc?id=yyy.` instead of using aria2c (you can comment that part out or delete it). 
(more on gdown [here](https://github.com/wkentaro/gdown)) 
Everything else will be the same. 
The download speeds were from 70MB/s all the way upto 200MB/s. Average being around 110MB/s. 
Now if I could only figure out how to use Localtunnel instead of using Cloudflare.  

I could edit the naifu.tar file and change the requirements.txt too but thats honestly too much work. Using the code above (ty) is much easier lol  
Screenshots:
**The code will look something like this**
![Screenshot (5)](https://user-images.githubusercontent.com/117362001/199786040-d595ecb2-c0ea-47f6-8ec6-c2c14a6093d2.png)

Naifu SS
![Screenshot (4)](https://user-images.githubusercontent.com/117362001/199786072-2e4c2fa1-d8d1-4aee-ad74-6e28e07ebd1b.png)

**Comment by lumibliss** [comment]
> I didn't make a backup of the github repo as this repo only stores the ipynb file, which you can just make a backup of on google drive itself. It's like 5kb or something. But what the other person did also works fine. I first used this `https://colab.research.google.com/drive/1LF2a4tY-bnyWJEbUeblv21nVuNDopNzQ` to download the two .tar files onto a brand new google account instead of downloading it and then uploading it manually. That notebook does it for you. Save it anywhere, it doesn't matter. Make the accessibility for naifu.tar and animefull-latest.tar to open so your colab can download it. Don't share it though, the file will get rate limited, just make your own copy. Then install gdown using `!pip install gdown` Then use gdown to download the file from your drive as following: `!gdown https://drive.google.com/u/2/uc?id=xxx ` `!gdown https://drive.google.com/u/2/uc?id=yyy.` instead of using aria2c (you can comment that part out or delete it). (more on gdown [here](https://github.com/wkentaro/gdown)) Everything else will be the same. The download speeds were from 70MB/s all the way upto 200MB/s. Average being around 110MB/s. Now if I could only figure out how to use Localtunnel instead of using Cloudflare.
> 
> I could edit the naifu.tar file and change the requirements.txt too but thats honestly too much work. Using the code above (ty) is much easier lol Screenshots: **The code will look something like this** ![Screenshot (5)](https://user-images.githubusercontent.com/117362001/199786040-d595ecb2-c0ea-47f6-8ec6-c2c14a6093d2.png)
> 
> Naifu SS ![Screenshot (4)](https://user-images.githubusercontent.com/117362001/199786072-2e4c2fa1-d8d1-4aee-ad74-6e28e07ebd1b.png)

You guys are amazing, thanks so much for sharing!!

I'm also interested to know, instead of using cloud flare link, to use own local host network...

If someone knows , please share

**Comment by manaheti1** [comment]
"instead of using cloud flare link, to use own local host network...'
Nope you are using the Google Collab host to run the script not the PC you are using, unless you download the script and run the code on your PC then you can use localhost network. But it not optimal.

**Comment by manaheti1** [comment]
> I didn't make a backup of the github repo as this repo only stores the ipynb file, which you can just make a backup of on google drive itself. It's like 5kb or something. But what the other person did also works fine. I first used this `https://colab.research.google.com/drive/1LF2a4tY-bnyWJEbUeblv21nVuNDopNzQ` to download the two .tar files onto a brand new google account instead of downloading it and then uploading it manually. That notebook does it for you. Save it anywhere, it doesn't matter. Make the accessibility for naifu.tar and animefull-latest.tar to open so your colab can download it. Don't share it though, the file will get rate limited, just make your own copy. Then install gdown using `!pip install gdown` Then use gdown to download the file from your drive as following: `!gdown https://drive.google.com/u/2/uc?id=xxx ` `!gdown https://drive.google.com/u/2/uc?id=yyy.` instead of using aria2c (you can comment that part out or delete it). (more on gdown [here](https://github.com/wkentaro/gdown)) Everything else will be the same. The download speeds were from 70MB/s all the way upto 200MB/s. Average being around 110MB/s. Now if I could only figure out how to use Localtunnel instead of using Cloudflare.
> 
> I could edit the naifu.tar file and change the requirements.txt too but thats honestly too much work. Using the code above (ty) is much easier lol Screenshots: **The code will look something like this**

Wow I didn't know google drive can download that fast, Thank you. Now I can use NovelAI until the end of time.

**Comment by hahaxddd18** [comment]
> Wow I didn't know google drive can download that fast, Thank you. Now I can use NovelAI until the end of time.

No worries, I only learnt about it when I was using koboldai.  

Hopefully this repo wont dmca'd (it probably will)
EDIT: Maybe there's a slight problem lol....gdrive won't let me download the file again. I guess i'm being rate limited. 
Getting another host is probably a good solution for continuous usage.  

EDIT2: OR just do this instead of gdown as it will cause the file to get rate limited.
```
from google.colab import drive
drive.mount('/content/drive')
%cd /content/drive/My Drive/naifu/

!echo "Decompressing..."
!tar xf naifu.tar --directory /content/
!echo "Done."
```
There is no download bar but it downloads. Don't forget to remove the `rm` statement so it wont just delete it from your gdrive like it did for me cause i'm stupid. 
Basically code will look like this in the end: 
![final code](https://user-images.githubusercontent.com/117362001/199815348-4deb4da9-1dcc-44d4-980b-0e2b116fd9e9.png)
Much safer than whatever the hell I said earlier and it makes much more sense too. Although the code is still hacky lol

**Comment by lumibliss** [comment]
What is the difference between the NAIFU file and anime latest? Do we really need them both?

**Comment by Redir1** [comment]
> > Hello, is it possible make ai draw better hands? also any way to make nai work with stable 1.5? or it's already updated?
> > put "bad hand" into Undesired tag. try looking on reddit NovelAI thread for prompts like "finely detail beautiful hand" like that, maybe it will help.
> > This version is the leak version and not up to date with current version running on NovelAI official website.
> > "work with stable 1.5" you mean running diffusionAI with NovelAI models? It just existed yesterday but the result is terrible so the creator delete it.

Damn, sad to hear, will leaked version gets update someday?

**Comment by slooi** [comment]
> I don't think I can fix this. You should wait for slooi or you can check this (ignore the video check the comment by Shahan Mohammad, they got the similar problem to us, and some people actually got it) https://www.youtube.com/watch?v=TyIL7XjMUaw&lc=UgznDfJ-al5qQlLMKWB4AaABAg ![image](https://user-images.githubusercontent.com/65333593/199554290-ad40910e-1a1c-4d4f-a59c-2d5fbd14b5d0.png)
I recently updated the repo with that solution

**Comment by DABLONKIE** [comment]
help how do i fix this

![Screenshot_657](https://user-images.githubusercontent.com/94186808/200089295-530fc711-f991-4e03-a452-f241a713b1e4.png)
